nixos/lib/make-options-doc/mergeJSON.py

```diff
@@ -41,6 +41,49 @@ def unpivot(options: Dict[Key, Option]) -> Dict[str, JSON]:
         result[opt.name] = opt.value
     return result
 
+# converts in-place!
+def convertMD(options: Dict[str, Any]) -> str:
+    import mistune
+    from xml.sax.saxutils import escape
+
+    def apply(node) -> str:
+        if node['type'] == 'paragraph':
+            return "".join(map(apply, node['children']))
+        elif node['type'] == 'text':
+            return escape(node['text'])
+        elif node['type'] == 'codespan':
+            return f"<literal>{escape(node['text'])}</literal>"
+        elif node['type'] == 'block_code':
+            return f"<programlisting>{escape(node['text'])}</programlisting>"
+        elif node['type'] == 'link':
+            if node['link'][0:1] == '#':
+                attr = "linkend"
+                link = node['link'][1:]
+            else:
+                attr = "xlink:href"
+                link = escape(node['link'])
+            return f"<link {attr}=\"{link}\">{''.join(map(apply, node['children']))}</link>"
+        else:
+            raise NotImplementedError("md node not supported yet", node)
+
+    for (name, option) in options.items():
+        if 'description' not in option:
+            continue
+        if type(option['description']) != dict:
+            continue
+        if '_type' not in option['description']:
+            continue
+        if option['description']['_type'] != 'mdDoc':
+            continue
+        text = option['description']['text']
+        ast = mistune.markdown(text, renderer='ast')
+        # pass the double line break for postprocess-option-descriptions.xsl.
+        # also keep trailing spaces so we can diff the generated XML to check for
+        # conversion bugs.
```

Comment:
Diffing seems like a good idea.

Instead of trying to match every quirk, we can choose to canonicalize both inputs to the diff. We'll need something like that for `<filename>` anyway.